src/codec.ts

```diff
@@ -30,6 +30,23 @@ export interface Field {
   value: Value;
 }
 
+export interface IProtoMessageParams {
```

Comment:
Curious why we are calling this IProtoMessageParams and not just ProtoMessageParams?

Reply:
It's a convention to declare interfaces but it looks we are not following it much on the handwritten side.  The gapic auto-generated protos also follow similar convention: https://github.com/googleapis/nodejs-spanner/blob/main/protos/protos.d.ts#L224